plumbing/transport/registry.go

```diff
@@ -0,0 +1,39 @@
+package transport
+
+import (
+	"fmt"
+	"sync"
+)
+
+// registry are the protocols supported by default.
+var (
+	registry = map[string]Transport{}
+	mtx      sync.Mutex
+)
+
+// Register adds or modifies an existing protocol.
+func Register(protocol string, c Transport) {
+	mtx.Lock()
+	defer mtx.Unlock()
+	registry[protocol] = c
```

Comment:
When the code is this simple it is not worth the additional costs of a `defer`.

```suggestion
	registry[protocol] = c
	mtx.Unlock()
```